### PR TITLE
reference/configuration: add configuration for reserver space

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

### What is changed, added or deleted? 

Add configuration for reserving space for TiKV.

### What is the related PR or file link(s)? 

- [Reserve disk space before create db engine ](https://github.com/tikv/tikv/pull/6321)

### Which TiDB version(s) does your changes apply to? 

- [x] v4.0 (TiDB 4.0 versions)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [ ] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

**Comment by TomShawn** [comment]
@Little-Wallace Please sign the CLA and involve a technical review, thanks!

**Comment by Little-Wallace** [comment]
@zhangjinpeng1987

File: dev/reference/configuration/tikv-server/configuration-file.md [hunk 1]
```diff
@@ -212,6 +212,12 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：100MB
 + 单位: MB|GB
 
+### `reserve-space`
+
++ TiKV 启动时预占额外空间的临时文件大小，临时文件名为`space_placeholder_file`，位于 storage.data-dir 目录下，TiKV 磁盘空间耗尽无法正常启动需要紧急干预时，可以删除该文件，并且将该项配置设置为 0
```
Comment (zhangjinpeng1987) [inline dev/reference/configuration/tikv-server/configuration-file.md]: Which config user can change?

Comment (TomShawn) [inline dev/reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ TiKV 启动时预占额外空间的临时文件大小。临时文件名为 `space_placeholder_file`，位于 `storage.data-dir` 目录下。TiKV 磁盘空间耗尽无法正常启动需要紧急干预时，可以删除该文件，并且将该项配置设置为 `0`。
```

Comment (lilin90) [reply dev/reference/configuration/tikv-server/configuration-file.md]: @zhangjinpeng1987 `reserve-space`, I think.

File: reference/configuration/tikv-server/configuration-file.md [hunk 2]
```diff
@@ -212,6 +212,12 @@ Coprocessor 线程池中线程的栈大小，默认值：10，单位：KiB|MiB|G
 + 默认值：100MB
 + 单位: MB|GB
 
+### `reserve-space`
+
++ TiKV 启动时预占额外空间的临时文件大小。临时文件名为 `space_placeholder_file`，位于 `storage.data-dir` 目录下。TiKV 磁盘空间耗尽无法正常启动需要紧急干预时，可以删除该文件，并且将reserve-space设置为 `0MB`。
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md]:
```suggestion
+ TiKV 启动时预占额外空间的临时文件大小。临时文件名为 `space_placeholder_file`，位于 `storage.data-dir` 目录下。TiKV 磁盘空间耗尽无法正常启动需要紧急干预时，可以删除该文件，并且将 `reserve-space` 设置为 `0MB`。
```

**Pull request merged**